### PR TITLE
[#38] As the application, I have the add-on for each template variant

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolves https://github.com/nimblehq/infrastructure-templates/issues/38

## What happened 👀

- Move AWS parts to add-on files.
- Change classes to functions to use easier.
- Add `postProcess` function to run some commands e.g. format code, detect terraform version.
- Refactor file helpers to accept only `projectName` arg as we just need it only.
- Remove redundant packages.

## Insight 📝

- With this, I think we can add tests for each add-on easier. It looks neat and nicer to separate
- You can try it by checking out this branch and running `bin/dev generate nimble-test`

## Proof Of Work 📹

Tests should be still passed

index_ts_—_infrastructure-templates

### CODE REVIEW

File: src/commands/generate/index.ts [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
+    name: 'provider',
+    message: 'Which cloud provider would you like to use?',
+    choices: [
+      {
+        value: 'aws',
+        name: 'AWS',
+      },
+      {
+        value: 'gcp',
+        name: 'GCP',
```
Comment (malparty) [inline src/commands/generate/index.ts:25]:
```suggestion
        name: 'GCP (NOT IMPLEMENTED YET)',
```
Same for Heroku 🙇 
It was quite frustrating when I tried this template and had to start the whole command again as I selected a non implemented option :D

Comment (hoangmirs) [reply src/commands/generate/index.ts]: 🙇 Added in https://github.com/nimblehq/infrastructure-templates/pull/70/commits/171467de4b5e65dfede751df80398d1d6f33ed91
I also disable those options 😄

**Comment by malparty** [comment]
It seems that the test failed, but as the code coverage returned a valid exit code, it has not been marked as failed :harold: 

From [this action](https://github.com/nimblehq/infrastructure-templates/runs/7426347705?check_suite_focus=true)

image


image

**Review by malparty — APPROVED** [review]
LGTM :)

**Review by andyduong1920 — APPROVED** [review]

**Pull request merged**